Formula/sevenzip.rb

```diff
@@ -30,6 +30,11 @@ def install
         ["gcc", "g"]
       end
 
+      if OS.mac?
+        system "sed", "-i", "''", "s/sysmacros.h/types.h/", "../../Common/FileStreams.cpp"
+        system "sed", "-i", "''", "s/sysmacros.h/types.h/", "../../UI/Common/UpdateCallback.cpp"
+      end
```

Comment:
Use `inreplace` instead of shelling out to `sed`. Also, please add a comment linking to the upstream report.